crates/partition-store/src/partition_store_manager.rs

```diff
@@ -42,14 +48,12 @@ pub enum OpenMode {
 
 #[derive(Clone, Debug)]
 pub struct PartitionStoreManager {
-    lookup: Arc<Mutex<PartitionLookup>>,
+    partition_stores: Arc<Mutex<PartitionStoreLookup>>,
+    persisted_lsns: Arc<PersistedLsnLookup>,
     rocksdb: Arc<RocksDb>,
 }
 
-#[derive(Default, Debug)]
-struct PartitionLookup {
-    live: BTreeMap<PartitionId, PartitionStore>,
-}
+type PartitionStoreLookup = HashMap<PartitionId, PartitionStore>;
```

Comment:
Switched to `ahash::HashMap` for performance; I don't think we rely on ordering anywhere as the `get_all_partition_stores`method below was unused.

Replacing the struct with a type seemed worthwhile at the same time.